src/storage/core.rs

```diff
@@ -413,54 +416,80 @@ where
     /// Returns entries with matching key
     /// # Errors
     /// Fails after any disk IO errors.
-    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<Vec<Entry>> {
+    pub async fn read_all(&self, key: impl AsRef<K>) -> Result<ReadResult<Vec<Entry>>> {
         let key = key.as_ref();
         let mut all_entries = Vec::new();
         let safe = self.inner.safe.read().await;
+        let blobs = safe.blobs.read().await;
```

Comment:
Looks like it is better to acquire `read` lock not here, but below, just right before actual access to `blobs` collection